.gitlab-ci.yml

```diff
@@ -76,6 +44,12 @@ cmake_build:
     CCACHE_BASEDIR: ${CI_PROJECT_DIR}/nrn
     CCACHE_DIR: ${CI_PROJECT_DIR}/ccache
   script:
+    - python3 -m virtualenv venv
+    - venv/bin/pip install --upgrade pip -r nrn_requirements.txt
```

Comment:
Can we keep it as a separate state for logs?

Reply:
Discussed over chat, but because virtualenvs break when you move them it's a tradeoff between one bigger stage with no custom runner config and the ability to scale to more runners, or separate jobs with custom runner config to ensure all jobs in a pipeline run in the same directory.